### ISSUE TITLE
Upgrade to `3.1.0-incubating.2` causes ClassCastException

### ISSUE THREAD

**Issue by metasim** [mpollmeier/gremlin-scala#142]
I haven't dug into this too much, but wanted to post this in case there was a know solution. I just upgraded from `3.1.0-incubating` to `3.1.0-incubating.2`, and am now getting exceptions of the following form for one of my traversals. Not sure if this is a bug on the TinkerPop side or gremlin-scala, but wanted to start here in case you had some advice.

```
org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerEdge cannot be cast to java.lang.Comparable
java.lang.ClassCastException: org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerEdge cannot be cast to java.lang.Comparable
	at org.apache.tinkerpop.gremlin.process.traversal.Order$1.compare(Order.java:33)
	at org.apache.tinkerpop.gremlin.process.traversal.step.map.OrderGlobalStep$ComparatorTraverser.compare(OrderGlobalStep.java:129)
	at org.apache.tinkerpop.gremlin.process.traversal.step.map.OrderGlobalStep$ComparatorTraverser.compare(OrderGlobalStep.java:119)
	at java.util.Comparator.lambda$thenComparing$36697e65$1(Comparator.java:216)
	at org.apache.tinkerpop.gremlin.util.function.ChainedComparator.compare(ChainedComparator.java:42)
	at java.util.Arrays.mergeSort(Arrays.java:1544)
	at java.util.Arrays.mergeSort(Arrays.java:1555)
	at java.util.Arrays.mergeSort(Arrays.java:1555)
	at java.util.Arrays.mergeSort(Arrays.java:1555)
	at java.util.Arrays.mergeSort(Arrays.java:1555)
	at java.util.Arrays.mergeSort(Arrays.java:1555)
	at java.util.Arrays.mergeSort(Arrays.java:1555)
	at java.util.Arrays.mergeSort(Arrays.java:1555)
	at java.util.Arrays.legacyMergeSort(Arrays.java:1523)
	at java.util.Arrays.sort(Arrays.java:1510)
	at java.util.ArrayList.sort(ArrayList.java:1454)
	at java.util.Collections.sort(Collections.java:175)
	at org.apache.tinkerpop.gremlin.process.traversal.traverser.util.TraverserSet.sort(TraverserSet.java:144)
	at org.apache.tinkerpop.gremlin.process.traversal.step.map.OrderGlobalStep.barrierConsumer(OrderGlobalStep.java:54)
	at org.apache.tinkerpop.gremlin.process.traversal.step.util.CollectingBarrierStep.processAllStarts(CollectingBarrierStep.java:60)
	at org.apache.tinkerpop.gremlin.process.traversal.step.util.CollectingBarrierStep.processNextStart(CollectingBarrierStep.java:75)
	at org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext(AbstractStep.java:140)
	at org.apache.tinkerpop.gremlin.process.traversal.step.util.ExpandableStepIterator.next(ExpandableStepIterator.java:54)
	at org.apache.tinkerpop.gremlin.process.traversal.step.map.FlatMapStep.processNextStart(FlatMapStep.java:46)
	at org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext(AbstractStep.java:140)
	at org.apache.tinkerpop.gremlin.process.traversal.step.util.ExpandableStepIterator.next(ExpandableStepIterator.java:54)
	at org.apache.tinkerpop.gremlin.process.traversal.step.sideEffect.SideEffectStep.processNextStart(SideEffectStep.java:38)
	at org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.next(AbstractStep.java:126)
	at org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.next(AbstractStep.java:37)
	at org.apache.tinkerpop.gremlin.process.traversal.Traversal.iterate(Traversal.java:166)
	at org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal.iterate(GraphTraversal.java:1230)
	at org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal$Admin.iterate(GraphTraversal.java:173)
	at org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.DefaultGraphTraversal.iterate(DefaultGraphTraversal.java:45)
	at gremlin.scala.GremlinScala.iterate(GremlinScala.scala:43)

```

**Comment by metasim** [comment]
Here's a test case that triggers it.

https://github.com/metasim/gremlin-scala/commit/9e568ad0a6be3d08fab70478a64e76d7af991828

In the mean time, I'll try to port my comparisons to use the `.orderBy()` methods instead.

**Comment by mpollmeier** [comment]
is this fixed by #143 ?

**Comment by metasim** [comment]
Fixed in #143
